### DISCUSSION TITLE
Are Filepond IDs lost during authenticating process?

### DISCUSSION THREAD

**Discussion by ludoguenet** [rahulhaque/laravel-filepond#39] [Q&A] [answered]
Hello 👋

I'm creating a Laravel 10 application where a guest uploads pictures through the server endpoint which is working.

Nevertheless, when the user logs in (the page is not refreshed) the restoring point (which also works as guest) can no longer retrieve files from IDs : filepond models are null.

Is it a normal behaviour? 

Thank you for reading me!

**Comment by rahulhaque** [comment]
@ludoguenet can you please elaborate a bit. Or share the problem code for checking.

**Comment by ludoguenet** [comment]
Certainly!

I use Vue Single File Components to handle front-end logic. I created a multi-step form that retains Filepond file IDs throughout the process. There's a specific step where users can submit photos, and I followed the documentation to set it up correctly.

At the end of the process, I use `Filepond::field($picturesIds)->getFile()` to store the actual files using Spatie Media Library, which works well.

However, I noticed that when users log in (or register and then log in), the file IDs are correctly preserved, but the pictures that should be retrieved when returning to the "photos" step are missing. The restore endpoints, which should return 200, instead return 500 with the following error message:

TypeError: RahulHaque\Filepond\AbstractFilepond::createFileObject(): Argument #1 ($filepond) must be of type RahulHaque\Filepond\Models\Filepond, null given. This error occurs in ..../vendor/rahulhaque/laravel-filepond/src/AbstractFilepond.php on line 161.

Many thanks for taking the time to assist me!

**Comment by rahulhaque** [comment]
@ludoguenet thanks for the explanation. I believe you're talking about filepond's `restore` endpoints described [here](https://pqina.nl/filepond/docs/api/server/#restore). If that's the case, I am sorry that this is not yet implemented. You can find the supported endpoints listed in the feature section of my doc. I am eager to implement this whenever possible. However you can write your own `restore` endpoint logic as well.

Thank you for using my package.

**Reply by ludoguenet** [reply]
I indeed am referring to the `restore` endpoints, which I have successfully implemented. It is currently functioning as expected: whenever I return to the "photos" step, they are restored based on their corresponding IDs.

However, a problem arises when the user proceeds to the "Register Or Login" step, logs in (using Fortify for authentication), and then navigates back to the "photos" step. At this point, the photos can no longer be restored. Nevertheless, the files themselves remain present in the temporary folder.

Here is the implementation of my restore functionality:

```
// web.php

Route::get(
    uri: config('filepond.server.url', '/filepond') . '/restore/{id}',
    action: RestoreFilepondController::class,
)->name('filepond.restore');
```

```
// RestoreFilepondController.php

final class RestoreFilepondController extends Controller
{
    public function __invoke(
        string $id,
    ): void {
        /**
         * @var UploadedFile $file
         */
        $file = Filepond::field($id)->getFile();

        header('Access-Control-Expose-Headers: Content-Disposition');
        header('Content-Type: ' . $file->getMimeType());
        header('Content-Disposition: inline; filename="' . $file->getClientOriginalName() . '"');

        echo $file->getContent();
    }
}
```

PS: You are welcome to use this code as a reference to help you build the restore feature :)

**Reply by rahulhaque** [reply]
Can you please try after removing `owned()` scope from [setFieldModel](https://github.com/rahulhaque/laravel-filepond/blob/748cf684c575d1614f89345462eb9b2f20dbfccd/src/AbstractFilepond.php#L101) function. Also, thanks a lot for the restore endpoint logic.

Edit: If it does, I will release a new version to handle this type of scenario. Also there are some endpoints I intentionally skipped from including in my package as their use cases may vary among users. `restore` endpoint is one of those endpoints that exposes a bit more of the system to end user than intended.

**Comment by ludoguenet** [comment]
It works like a charm! :)

**Reply by rahulhaque** [reply, marked as answer]
@ludoguenet  thanks for opening the discussion and pointing out a real world scenario for my package. I've released a new version [v1.10.10](https://github.com/rahulhaque/laravel-filepond/releases/tag/v1.10.10) with the option to force skip ownership check for special cases like yours. Update to the latest release get the file like this in your `restore` endpoint logic - 
```php
$file = Filepond::field($id, false)->getFile();
```
Let me know of the result after the test.

**Reply by ludoguenet** [reply]
I tested the actual scenario and it is working well! Thank you very much!

**Reply by rahulhaque** [reply]
I see you are content creator and doing decent job for the PHP Laravel community. If you ever happen to include my package in any of your tutorial videos, please do let me know. I would love to attach the link in my documentation. I appreciate the effort you put to teach others. Thank you.

**Reply by ludoguenet** [reply]
I would be delighted to, I actually wanted to suggest it to you. The only drawback is that my tutorials are in French.

**Reply by rahulhaque** [reply]
I think, with the subtitle and auto translate feature of YT, language is not a barrier anymore. I would gladly keep your link in the tutorial section even if there's a English tutorial comes after. Yours will be the first to make entry. Thank you for considering the idea.

**Reply by ludoguenet** [reply]
Done! Here's the link: https://youtu.be/IQ3fEseDck8. Although it's not published yet, the link should remain unchanged, so feel free to add it now if you find it decent enough!

**Reply by rahulhaque** [reply]
Thank you sooo much. I'll add this right away. 🫶

**Reply by ludoguenet** [reply]
Thank you for creating such an amazing library. 😊
